src/System.Device.Gpio/Interop/Unix/libgpiod/V2/Proxies/LibGpiodProxyBase.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// Disable these StyleCop rules for this file, as we are using native names here.
+#pragma warning disable SA1300 // Element should begin with upper-case letter
+
+namespace System.Device.Gpio.Interop.Unix.libgpiod.V2.Proxies;
+
+internal abstract class LibGpiodProxyBase
+{
+    /// <summary>
+    /// Each proxy instance gets its dedicated lock instance.
+    /// </summary>
+    private readonly object _lock = new();
+
+    /// <summary>
+    /// Helper function that wraps any exception (from a native call) in <see cref="GpiodException"/> for easier exception handling on client side.
+    /// </summary>
+    protected static void TryCallGpiod(Action action)
+    {
+        try
+        {
+            action.Invoke();
+        }
+        catch (Exception e) when (e is not GpiodException)
+        {
+            throw new GpiodException("Exception while calling libgpiod", e);
```

Comment:
I believe that this may cause issues with the lock. It is convention that `TryXxx` methods  do not throw and for them to instead return `bool success`. This is specially important here, since you are calling these methods inside a lock statement, so you have to be careful of those methods not handling exceptions correctly as you could end up in an inconsistent state or unanticipated behavior.

Reply:
cc: @stephentoub can you please take a look at this class? The TL;DR is that sub types of this base will want to have a lot of its methods to not execute concurrently, so to protect against it this is introducing a lock and some helper methods (TryCallGpiodLocked) that will try to ensure that these methods get executed in a thread-safe manner. I'm sure this is a common pattern, do we have precedent for suggestions for these cases?

Reply:
For clarification: libgpiodv2 is thread aware but not thread safe. That means libgpiod objects may be accessed from any thread but only one at a time. Also I found it useful to wrap any exception, to not require much exception handling on caller side, and bloating docs that has to list every possible exception.
This base class implements these aspects, for code use. I do not see any issue regarding locks.
I am open to rename the methods though to not clash with the Try pattern :)

Reply:
As it turns out reading and writing concurrently is in most cases not a problem see [conversation](https://lore.kernel.org/linux-gpio/CAEydidkEikZciGD_EzS95tMXznPH=OFP__Q2Bkk6HrQ+sNhkxQ@mail.gmail.com/T/#t)
Changed it so only writes are now locked.

Reply:
@joperezr The TryXXX methods are now renamed, and locks are only held on reads. Is it ok for you to resolve this?